# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fs/child/FileSystemDirectoryIteratorFactory.cpp
# Commit: 214c4d6bf7f9
# Full Hash: 214c4d6bf7f98a332c4ab81d325d9245ae810955
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2022-10-20 21:51:26
# Regressor Bug: 1792245
# File Overlap Count: 2
# Description:
#   Bug 1792245 - Add ErrorResult to FileSystemRequestHandler methods; r=dom-storage-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D159572
# ==============================================================================

diff -r 330c69218a93 -r 214c4d6bf7f9 dom/fs/child/FileSystemDirectoryIteratorFactory.cpp
--- a/dom/fs/child/FileSystemDirectoryIteratorFactory.cpp	Thu Oct 20 07:33:41 2022 +0300
+++ b/dom/fs/child/FileSystemDirectoryIteratorFactory.cpp	Thu Oct 20 04:21:30 2022 +0000
@@ -108,7 +108,10 @@
       return nullptr;
     }
 
-    next(aGlobal, aManager, promise);
+    next(aGlobal, aManager, promise, aError);
+    if (aError.Failed()) {
+      return nullptr;
+    }
 
     return promise.forget();
   }
@@ -117,7 +120,7 @@
 
  protected:
   void next(nsIGlobalObject* aGlobal, RefPtr<FileSystemManager>& aManager,
-            RefPtr<Promise> aResult) {
+            RefPtr<Promise> aResult, ErrorResult& aError) {
     MOZ_ASSERT(aResult);
 
     Maybe<DataType> rawValue;
@@ -126,10 +129,8 @@
     // we hit the end of a page?
     // How likely it is that it would that lead to wasted fetch operations?
     if (0u == mWithinPageIndex) {
-      ErrorResult rv;
-      RefPtr<Promise> promise = Promise::Create(aGlobal, rv);
-      if (rv.Failed()) {
-        aResult->MaybeReject(std::move(rv));
+      RefPtr<Promise> promise = Promise::Create(aGlobal, aError);
+      if (aError.Failed()) {
         return;
       }
 
@@ -158,14 +159,16 @@
               nextInternal(value);
             }
 
-            IgnoredErrorResult rv;
             ResolveValue(global, manager, value, aResult);
           },
           [](nsresult aRv) {});
       promise->AppendNativeHandler(listener);
 
       FileSystemRequestHandler{}.GetEntries(aManager, mEntryId, mPageNumber,
-                                            promise, newPage);
+                                            promise, newPage, aError);
+      if (aError.Failed()) {
+        return;
+      }
 
       ++mPageNumber;
       return;